text/0000-clippy-uno.md

```diff
@@ -0,0 +1,721 @@
+
+ - Adding entirely new lints
+ - Fixing false positives (A lint may no longer lint in a buggy case)
+ - Fixing false negatives (A case where the lint _should_ be linting but doesn’t is fixed)
+ - Bugfixes (When the lint panics or does something otherwise totally broken)
+
+When fixing false negatives this will usually be fixing things that can be
+understood as comfortably within the scope of the lint as documented/named.
+For example, a lint on having the type `Box<Vec<_>>` may be changed to also catch `Box<Vec<T>>`
+where `T` is generic, but will not be changed to also catch `Box<String>` (which can be linted
+on for the same reasons).
+
+An exception to this is the "nursery" lints &mdash; Clippy has a lint category for unpolished lints called the "nursery" which
+are allow-by-default. These may experience radical changes, however they will never be entirely "removed" either.
+
+Pre-1.0 we may also flush out all of the deprecated lints.
+
+## Lint audit and categories
+
+A couple months ago we did a lint audit to recategorize all the Clippy lints. The [Reference-Level explanation below][cat] contains a list
+of all of these lints as currently categorized.
+
+The categories we came up with are:
+
+
+ - Correctness (Deny): Probable bugs, e.g. calling `.clone()` on `&&T`,
+   which clones the (`Copy`) reference and not the actual type
+ - Style (Warn): Style issues; where the fix usually doesn't semantically change the code but instead changes naming/formatting.
+   For example, having a method named `into_foo()` that doesn't take `self` by-move
+ - Complexity (Warn): For detecting unnecessary code complexities and helping
+   simplify them. For example, a lint that asks you to replace `.filter(..).next()` with `.find(..)`
+ - Perf (Warn): Detecting potential performance footguns, like using `Box<Vec<T>>` or calling `.or(foo())` instead of `or_else(foo)`.
+ - Pedantic (Allow): Controversial or exceedingly pedantic lints
+ - Nursery (Allow): For lints which are buggy or need more work
+ - Cargo (Allow): Lints about your Cargo setup
+ - Restriction (Allow): Lints for things which are not usually a problem, but may be something specific situations may dictate disallowing.
+ - Internal (Allow): Nothing to see here, move along
+ - Deprecated (Allow): Empty lints that exist to ensure that `#[allow(lintname)]` still compiles after the lint was deprecated.
+
+Lints can only belong to one lint group at a time, and the lint group defines the lint level. There is a bunch of overlap between
+the style and complexity groups -- a lot of style issues are also complexity issues and vice versa. We separate these groups
+so that people can opt in to the complexity lints without having to opt in to Clippy's style.
+
+## Compiler uplift
+
+The compiler has historically had a "no new lints" policy, partly with the desire that lints would
+incubate outside of Clippy. This feels like a good time to look into uplifting these lints.
```

Comment:
did you mean, perhaps: "outside of the compiler, in Clippy?" (or some variant there of) - reads backwards currently (or I've misunderstood).